src/hotspot/cpu/x86/stubGenerator_x86_64_arraycopy.cpp

```diff
@@ -965,6 +1043,58 @@ void StubGenerator::arraycopy_avx3_special_cases(XMMRegister xmm, KRegister mask
   __ jmp(L_exit);
 }
 
+void StubGenerator::arraycopy_avx3_special_cases_256(XMMRegister xmm, KRegister mask, Register from,
+                                                     Register to, Register count, int shift, Register index,
+                                                     Register temp, Label& L_entry, Label& L_exit) {
+  Label L_entry_64, L_entry_128, L_entry_192, L_entry_256;
+
+  int size_mat[][4] = {
+  /* T_BYTE */ {64, 128, 192, 256},
+  /* T_SHORT*/ {32, 64 , 96 , 128},
+  /* T_INT  */ {16, 32 , 48 ,  64},
+  /* T_LONG */ { 8, 16 , 24 ,  32}
+  };
+
+  if (MaxVectorSize == 64) {
```

Comment:
This should be an assert here instead of if check as this method shouldn't be called if MaxVectorSize is < 64.

Reply:
Thanks, done.